### PR TITLE
[Cleanup] Cleans Controller Scaffold

### PR DESCRIPTION
**Pull request by eliasjpr**
Issue: https://github.com/amberframework/amber/issues/851

### Description of the Change

Granite adds some nice features that allows controllers to be thinner
and cleaner. With these changes controllers becomes almost on par with
Rails controller approach.

### Benefits

* Removes conditional branches
* Uses before_action filters
* Users Granite `find!`
* Laverages on Granite save to run validations instead of calling
`valid?` and then `save`

### Expected Behavior

- We expect behavior to be the same.

### CODE REVIEW

File: src/amber/cli/templates/scaffold/controller/granite/src/controllers/{{name}}_controller.cr.ecr [hunk 1]
```diff
@@ -1,74 +1,61 @@
...
-
-    if <%= @name %>.valid? && <%= @name %>.save
-      flash["success"] = "Created <%= class_name %> successfully."
-      redirect_to "/<%= Inflector.pluralize(@name) %>"
-    else
-      flash["danger"] = "Could not create <%= class_name %>!"
-      render("new.<%= @language %>")
-    end
+  def edit
+    render("edit.slang")
```
Comment (drujensen) [inline src/amber/cli/templates/scaffold/controller/granite/src/controllers/{{name}}_controller.cr.ecr]: might not be `slang`

**Comment by eliasjpr** [comment]
Would be nice to have render take a flash object then we can clean these lines:

```crystal
flash["danger"] = "Could not update <%= class_name %>!"
render("edit.<%= @language %>")
```

**Comment by faustinoaq** [comment]
@eliasjpr Looks like we need to update generated specs here as well :sweat_smile:

**Comment by faustinoaq** [comment]
I just realize this PR depends on #849 

@eliasjpr I changed this code without checking that PR, my bad :sweat_smile:

**Comment by eliasjpr** [comment]
Weird because I push those fixes about the set method today

**Comment by faustinoaq** [comment]
@eliasjpr can you sync and push them again :sweat_smile:

**Pull request merged**